## 272: Re: [blupf90] automatic looping through traits

- Title: 272: Re: [blupf90] automatic looping through traits
- Author: Ignacio Aguilar
- Date: Mar 13, 2014
- Original: [https://groups.yahoo.com/neo/groups/blupf90/conversations/messages/272](https://groups.yahoo.com/neo/groups/blupf90/conversations/messages/272)

```
I think that this can be done with unix tools

given a parameter file one snp, you need to change the position of the trait in your parameter file, 
in the following example I change not the trait, but the column for one effect, which in the example is in the 14 th
row. 
in your case you need to change the row number 5 (corresponding to trait position) 


The awk sentence will put the column to iterate through SNP, 
in this file start is the column pos for first SNP and all are with space delimiters. 


par.b90 is a template parameter file to run
with tail you can extract the last n rows from log file and store it  in a file for all snps for later analyses. 


Ignacio 




#!/bin/bash


nsnp=46549


start=5
rm residual_var
for j in `seq 1 $nsnp`
do


if [ $(($j%100)) -eq 0 ]; then echo " @ snp: " $j; fi
i=$(($start+$j-1))
awk '{ if (NR==14) print '$i',	"    1	   cov"; else print $0 }' par.b90 > pari.b90


echo pari.b90 | remlf90 > log


a=`tail -2 log | head -1 | tr "\n" " "`
echo $a >> residual_var


done


On 13/03/2014, at 11:10, Muir, William M. <bmuir@...> wrote:



I am trying the check quality of SNPs by their heritability, i.e. how well they follow the pedigree.  I think Andres
suggested this method? I advised a student in Agronomy to try this method on  a plant species he is working on.  It
worked very well, he tried this manually for about 100 SNPs, out of 11,000, and found that about 30% of the SNPs are
bad.  In plant species many have ancient whole genome duplications which results in what is called homeo SNPs. These
appear as SNPs in whole genome sequences but are actually SNPs between duplicates on different chromosomes and thus do
not behave in a Mendelian manner, much like a CNV.   The model is very simple, the ith SNP is the dependent variable
(coded 1,2,3) and the model contains the mean (fixed) and animal ID (random).  Since the student has a large number of
SNPs we are trying to find a way to automate the process.  I could write a Fortran program that calls REMLF90 for each
of the SNPs, but thought there might be a nifty scripting method in the UNIX command language that could do the same
thing?	I don't have much experience with UNIX (enough to be dangerous).  Does anyone have suggestions?  Thanks much
for your help.	Bill	
```

- [271](0271.md): automatic looping through traits by Muir, William M., Mar 13, 2014
    - [272](0272.md): Re: [blupf90] automatic looping through traits by Ignacio Aguilar, Mar 13, 2014
        - [275](0275.md): RE: [blupf90] automatic looping through traits by Ignacy Misztal, Mar 13, 2014
            - [276](0276.md): Re: [blupf90] automatic looping through traits by Andres Legarra, Mar 14, 2014
    - [274](0274.md): Re: [blupf90] automatic looping through traits by Andres Legarra, Mar 13, 2014
