## 389: DGV in postGSF90 for prediction of SNP effects

- Title: 389: DGV in postGSF90 for prediction of SNP effects
- Author: Ignacy Misztal
- Date: Oct 27, 2014

```
postGSF90 used GEBV for conversion to SNP effects. With the following 
 option:
 OPTION snp_effect_dgv
 the conversion uses DGV (direct genomic value)

 The quality of prediction with DGV is higher and especially so when 
 genotypes are on low-accuracy animals. In general:
 GEBV = w1 PA+ w2PC + w3CD + w4 DGV - w5 PI
 and only DGV is due to the genomic information.

 Daniela, Shogo and I figured a need for the change when analyzing Angus 
 commercial data. Now, indirect prediction via SNP effects is almost as 
 good as complete GEBV. Ignacio changed the code and also added 
 capability to see some of the weights as in the Appendix of AGuilar et 
 al. (2010) paper.

 Ignacy
```

- [384](0384.md): Re: Inverse of the Average Information Matrix by Shogo Tsuruta, Sep 24, 2014
    - [388](0388.md): Re: Inverse of the Average Information Matrix by jukebop, Sep 24, 2014
        - [389](0389.md): DGV in postGSF90 for prediction of SNP effects by Ignacy Misztal, Oct 27, 2014
    - [387](0387.md): Re: [blupf90] Re: Inverse of the Average Information Matrix by Ignacy Misztal, Sep 24, 2014
    - [385](0385.md): Re: [blupf90] Re: Inverse of the Average Information Matrix by John Cole, Sep 24, 2014
